### PR TITLE
[VIRTS-1879] Better docs structure

### PR DESCRIPTION
**Pull request by b17zr**
## Description

- Added Basic Usage page and move most of the former "Learning the Terminology" content there
- Rewrote "Learning the Terminology" page
- Removed CALDERA 2.0 information
- Added "Troubleshooting" page, from the "Common problems" (I felt like this name implied something..) page

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [?] This change requires a documentation update

## How Has This Been Tested?

Regenerated docs with sphinx.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [?] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by b17zr** [comment]
Also updated the server configuration page to include startup parameters (VIRTS-1831) and a documented config file.

File: sphinx-docs/Basic-Usage.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
+
+# Basic Usage
+
+## Agents
+
+### Agent Management
+
+To deploy an agent:
+
+1. Navigate to the Agents tab and click the "Click here to deploy an agent" button5 n
```
Comment (bworrell) [inline sphinx-docs/Basic-Usage.md:10]: The `button5 n` at the end of the line looks out of place. Is that intentional?

File: sphinx-docs/Basic-Usage.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+### Repeatable Abilities and Planners
+
+When creating a new operation, selecting a profile with repeatable abilities will disable both the _atomic_ and the _buckets_ planners. Due to the behavior and functionality of these planners, repeatable abilities will result in the planner looping infinitely on the repeatable ability. It is recommended to use the _batch_ planner with profiles containing repeatable abilities.
+
+## Plugins
+
+CALDERA is built using a plugin architecture on top of the core system. Plugins are separate git repositories that plug new features into the core system. Each plugin resides in the plugins directory and is loaded into CALDERA by adding it to the local.yml file.
+
+PLugins can be added through the UI or in the configuration file (likely `conf/local.yml`). Changes to the configuration file while the server is shut down. The plugins will be enabled when the server restarts.
```
Comment (bworrell) [inline sphinx-docs/Basic-Usage.md:309]: `PLugin`

File: sphinx-docs/Basic-Usage.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+Payloads can be stored as regular files or you can xor (encode) them so the anti-virus on the server-side does not pick them up. To do this, run the app/utility/payload_encoder.py against the file to create an encoded version of it. Then store and reference the encoded payload instead of the original.
+
+> The payload_encoder.py file has a docstring which explains how to use the utility.
+
+Payloads also can be ran through a packer to obfuscate them further from detection on a host machine.  To do this you would put the packer module name in front of the filename followed by a colon ':'.  This non-filename character will be passed in the agent's call to the download endpoint, and the file will be packed before sending it back to the agent. UPX is currently the only supported packer, but adding addition packers is a simple task.
+
+> an example for setting up for a packer to be used would be editing the filename in the payload section of an ability file: - upx:Akagi64.exe
```
Comment (bworrell) [inline sphinx-docs/Basic-Usage.md:109]: `an` should be `An`

**Review by bworrell — CHANGES_REQUESTED** [review]
I pulled this down and ran `make clean html` in the sphinx docs folder and it built/rendered fine (loaded it up in my browser). I didn't go through this too carefully but only noticed a couple of small typos. Overall this looks like an improvement in the structure of the documentation.

**Comment by b17zr** [comment]
https://github.com/mitre/fieldmanual/pull/63 will conflict, but I've included the DNS config information in Server-Configuration.md.

**Review by bworrell — DISMISSED** [review]
LGTM! I pulled it down and rebuilt the docs locally and it all loaded up correctly for me.

File: sphinx-docs/Server-Configuration.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+Configuration file changes must be made while the server is shut down. Any changes made to the configuration file while the server is running will be overwritten.
+
+The YAML configuration file contains all the configuration variables CALDERA requires to boot up and run. A documented configuration file is below:
+
+```yaml
+ability_refresh: 60  # Interval at which ability YAML files will refresh from disk 
+api_key_blue: BLUEADMIN123  # API key which grants access to CALDERA blue
+api_key_red: ADMIN123  # API key which grants access to CALDERA red
+app.contact.dns.domain: mycaldera.caldera  # Domain for the DNS contact server
```
Comment (bworrell) [inline sphinx-docs/Server-Configuration.md:25]: It looks like some of the DNS configuration info may have gotten lost during the merge conflict resolution. Is it worth including some of the info from the original file? https://github.com/mitre/fieldmanual/pull/64/files#diff-8d8033dc12d18ad455b4300a0f7237f86700a4f8fd177e0cbd566b064272fc51L45-L46

Comment (b17zr) [reply sphinx-docs/Server-Configuration.md:25]: The only information I see missing is "The C2 server will respond with the NXDOMAIN response code for any other domain". It feels more appropriate to put this into a page or section dedicated to the different contact methods.

**Review by bworrell — APPROVED** [review]
LGTM after resolving merge conflicts

**Pull request merged**